pkg/controller/provider/container/ovirt/client.go

```diff
@@ -226,3 +234,365 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
+			time.Sleep(1 * time.Second)
+			currImageTransferReq, errIT := conn.SystemService().ImageTransfersService().ImageTransferService(imageTransferId).Get().Send()
+			if err != nil {
+				err = liberr.Wrap(errIT)
+				return
+			}
+			currImageTransfer, ok = currImageTransferReq.ImageTransfer()
+		} else {
+			break
+		}
+	}
+
+	caCert := secret.Data["cacert"]
+	caCertPool := x509.NewCertPool()
+	ok = caCertPool.AppendCertsFromPEM(caCert)
+	if !ok {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Create http client
+	tlsConfig := &tls.Config{
+		RootCAs: caCertPool,
+	}
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+	client := &http.Client{Transport: transport}
+
+	itURL, ok := currImageTransfer.ProxyUrl()
+
+	// creat regex expresions to filter the proxy URL
+	reInit := regexp.MustCompile(`https://`)
+	reUrl := regexp.MustCompile(`^[^/]+`)
+
+	transferURL := reInit.ReplaceAllString(itURL, "")
+	transferURL = reUrl.FindString(transferURL)
+	transferURL = fmt.Sprint("https://", transferURL, "/info/")
+
+	// Prepare disk GET request
+	req, err := http.NewRequest("GET", transferURL, nil)
+
+	// Run the request
+	resp, err := client.Do(req)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	defer resp.Body.Close()
+
+	//close image transfer
+	_, err = conn.SystemService().ImageTransfersService().ImageTransferService(imageTransferId).Cancel().Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		//retrive missing CA cert and update secrete
+		caUrl := fmt.Sprint(string(secret.Data["url"]), "ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA")
+		cmd := exec.Command("wget", "--no-check-certificate", "-O", "ca.pem", caUrl)
+		err = cmd.Run()
+		if err != nil {
+			err = liberr.Wrap(err)
+			return
+		}
+
+		data, errFile := ioutil.ReadFile("ca.pem")
+		if err != nil {
+			err = liberr.Wrap(errFile)
+			return
+		}
+
+		updatedCa := fmt.Sprint(string(secret.Data["caCert"]), "/n", string(data))
+
+		//create k8s client
+		config, errClient := rest.InClusterConfig()
+		if err != nil {
+			err = liberr.Wrap(errClient)
+			return
+		}
+
+		clientset, errConfig := kubernetes.NewForConfig(config)
+		if err != nil {
+			err = liberr.Wrap(errConfig)
+			return
+		}
+
+		secretsClient := clientset.CoreV1().Secrets("forklift-konveyor")
+
+		secret := &core.Secret{
+			Data: map[string][]byte{
+				"caCert":   []byte(updatedCa),
+				"url":      []byte(string(secret.Data["url"])),
+				"user":     []byte(string(secret.Data["user"])),
+				"password": []byte(string(secret.Data["password"])),
+			},
+			Type: "Opaque",
+		}
+		_, err = secretsClient.Update(context.TODO(), secret, metav1.UpdateOptions{})
+		if err != nil {
+			err = liberr.Wrap(err)
+			return
+		}
+		ok = false
+		return
+	}
+
+	ok = true
+	return
+
+}
+
+// func uploadDiskTestConection() bool {
```

Comment:
this is currently marked as a comment but this is the code for the uploaded flow, keeping it here for now just in case.